### ISSUE TITLE
Mesh node position/index wrong when FINITE=3

### ISSUE THREAD

**Issue by martijnende** [ydluo/qdyn#24]
When a simulation is run with `FINITE=3` (forced symmetry to simulate free surface), the resulting snapshot output does not maintain sorted order. The last mesh element in each snapshot has a position that should correspond with the first element. To illustrate this a bit better, I have plotted the mesh node position in a 1D fault against its index:

![position_bug](https://user-images.githubusercontent.com/6959003/51602171-9a452b80-1f06-11e9-8155-25bc981be5a5.png)

However, the stress/slip/velocity/etc. output do not exactly match the first element:

![slip_bug](https://user-images.githubusercontent.com/6959003/51602332-f3ad5a80-1f06-11e9-98b6-c9aac8eb3b26.png)

When plotting slip distributions, I get the following result (note discontinuity at the bottom):

![first_elem_bug_unsorted](https://user-images.githubusercontent.com/6959003/51602370-0c1d7500-1f07-11e9-9da4-bc50d7e5c89e.png)

Or when the elements are sorted by their position (so that last element becomes the first; note discontinuity at the top):

![first_elem_bug](https://user-images.githubusercontent.com/6959003/51602405-1dff1800-1f07-11e9-8bb5-2a0ff11f129b.png)

This problem does not emerge when `FINITE=1`. The (Python) code to reproduce the issue:

```
from pyqdyn import qdyn
import numpy as np
import matplotlib.pyplot as plt

p = qdyn()

N = np.power(2, 6)
t_yr = 3600*24*365.0
V = 1e-9

# Python dictionary with general settings
set_dict = {
    "N": N,
    "NXOUT": np.power(2, 1),
    "NTOUT": 100,
    "ACC": 1e-7,
    "MU": 30e9,
    "DTTRY": 1e-8,
    "TMAX": 200*t_yr,
    "MESHDIM": 1,
    "VS": 3000,
    "L": 1.0e3,
    "FINITE": 3,
    "IC": 0,
    "SOLVER": 1,
    "V_PL": V,
}

# Python dictionary with rate-and-state friction parameters
set_dict_RSF = {
    "RNS_LAW": 0,
    "THETA_LAW": 1,
    "DC": 10e-3,
    "V_0": V,
    "V_SS": V,
    "MU_SS": 0.6,
    "A": 1e-2,
    "B": 2e-2,
    "SIGMA": 50e6,
}

set_dict["TH_0"] = 0.99*set_dict_RSF["DC"]/set_dict_RSF["V_0"]
set_dict["SET_DICT_RSF"] = set_dict_RSF

p.settings(set_dict)
p.render_mesh()
p.write_input()
p.run()
```

**Comment by jpampuero** [comment]
Which subroutine outputs the "position" plotted in your first figure?

**Comment by jpampuero** [comment]
But function render_mesh in pyqdyn.py recomputes X (and fiferently than subroutine init_mesh_1D in mesh.f90). Which one are you plotting?

**Comment by martijnende** [comment]
Embarrassingly, the `pyqdyn.py` wrapper skipped a line while reading the snapshots, so that the last element in each snapshot is actually the first element in the next snapshot. I will create a pull request with the fix.

**Comment by jpampuero** [comment]
In addition to that, there is the inconsistency I pointed out: function render_mesh in pyqdyn.py recomputes X differently than subroutine init_mesh_1D in mesh.f90

**Comment by martijnende** [comment]
The positions of the nodes are now addressed with PR #37. I'm closing this issue
